pkg/api/mock/matcher/example_test.go

```diff
@@ -0,0 +1,98 @@
+package matcher
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.anx.io/go-anxcloud/pkg/api/mock"
+)
+
+// Object is a custom gomega matcher which is used to test for matching objects
+// on the MockAPI instance by comparing the type and provided field names
+func ExampleObject() {
+	defer GinkgoRecover()
+
+	a := mock.NewMockAPI()
+
+	if err := a.Create(context.TODO(), &clouddnsv1.Record{ZoneName: "example.com", Name: "webmail"}); err != nil {
+		log.Fatalf("failed to create DNS record: %s", err)
+	}
+
+	// a.Existing() retrieves all objects which aren't currently flagged as destroyed
+	ok := Expect(a.Existing()).To(
```

Comment:
In a real Ginkgo/Gomega test we don't use the return value, maybe we should just print "Success" after this line - if it doesn't succeed, the matcher panics and this is recovered by Gomega